src/Microsoft.Framework.ConfigurationModel.Interfaces/IConfigurationSourceRoot.cs

```diff
@@ -7,6 +7,8 @@ namespace Microsoft.Framework.ConfigurationModel
 {
     public interface IConfigurationSourceRoot : IConfiguration
     {
+        string BasePath { get; }
```

Comment:
`BasePath` would be available in all configuration sources because `IConfigurationSourceRoot` inherits from `IConfiguration`.  would need `SomeConfigurationSource(IConfiguration source, ...)` constructors and appropriate changes to the extension methods.